catalog/security/banner-marking/src/main/java/org/codice/alliance/security/banner/marking/BannerCommonMarkingExtractor.java

```diff
@@ -13,77 +13,119 @@
-        tempMap.put(SECURITY_RELEASABILITY, this::processReleasability);
-        tempMap.put(SECURITY_CODEWORDS, this::processCodewords);
-        tempMap.put(SECURITY_DISSEMINATION_CONTROLS, this::processDissem);
+    private void checkSecurityAttributes(String name, List<Serializable> oldVals,
+            List<Serializable> newVals) {
+        if (!isSecurityAttribute(name) || CollectionUtils.isEmpty(oldVals)) {
+            return;
+        }
 
-        // At the present moment, we are going to be filling the security.owner-producer and
-        // security.classification-system fields with the same content, the trigraph/tetragraph
-        // of the originating country/organization. That may diverge in the future
-        tempMap.put(SECURITY_OWNER_PRODUCER, this::processOwnerProducer);
-        tempMap.put(SECURITY_CLASSIFICATION_SYSTEM, this::processClassSystem);
+        Set<String> oldValues = new HashSet<>((List<String>) (List<?>) oldVals);
+        Set<String> newValues = new HashSet<>((List<String>) (List<?>) newVals);
+        if (!oldValues.equals(newValues)) {
+            String errorMessage = String.format(
+                    "Extracted security attribute %s from the banner markings"
+                            + " does not match value pre-existing on the metacard. The"
+                            + " metacard will not be created.",
+                    name);
+            INGEST_LOGGER.error(errorMessage);
+            throw new RuntimeException(errorMessage);
+        }
+    }
 
-        setAttProcessors(tempMap);
+    private boolean isSecurityAttribute(String attrName) {
+        return securityMap.keySet()
+                .contains(attrName);
     }
 
-    Attribute processClassMarking(Metacard metacard, BannerMarkings bannerMarkings) {
-        List<String> classifications = Lists.newArrayList();
-        Attribute currAttr = metacard.getAttribute(SECURITY_CLASSIFICATION);
+    public Attribute processClassMarking(Metacard metacard, BannerMarkings bannerMarkings) {
+        Attribute currAttr = metacard.getAttribute(Security.CLASSIFICATION);
+        String extractedClassValue = translateClassification(bannerMarkings.getClassification(),
+                bannerMarkings.isNato(),
+                bannerMarkings.getNatoQualifier());
         if (currAttr != null) {
-            classifications.add((String) currAttr.getValue());
+            checkSecurityAttribute(Security.CLASSIFICATION,
+                    currAttr.getValue(),
+                    extractedClassValue);
         }
-        classifications.add(
-                translateClassification(bannerMarkings.getClassification(), bannerMarkings.isNato(),
-                        bannerMarkings.getNatoQualifier()));
-        return new AttributeImpl(SECURITY_CLASSIFICATION,
-                ImmutableList.<String>copyOf(classifications));
+        return new AttributeImpl(Security.CLASSIFICATION, extractedClassValue);
```

Comment:
❓ This attribute changed from multi-valued to single-valued?

Reply:
Yes it's defined as single valued in the catalog taxonomy